<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
 "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  <meta http-equiv="Content-Style-Type" content="text/css" />
  <meta name="generator" content="pandoc" />
  <title>Presentation</title>
  <style type="text/css">
      code{white-space: pre-wrap;}
      span.smallcaps{font-variant: small-caps;}
      span.underline{text-decoration: underline;}
      div.column{display: inline-block; vertical-align: top; width: 50%;}
  </style>
  <link rel="stylesheet" type="text/css" media="screen, projection, print"
    href="https://www.w3.org/Talks/Tools/Slidy2/styles/slidy.css" />
  <script src="https://www.w3.org/Talks/Tools/Slidy2/scripts/slidy.js"
    charset="utf-8" type="text/javascript"></script>
</head>
<body>
<div id="extract-useful-data-from-rna-seq-results-in-wt-and-transgenic-tomato-fruits" class="slide section level4">
<h1>Extract useful data from RNA-seq results in WT and transgenic tomato fruits</h1>
<pre><code>                         Haiping Liu Ph.D</code></pre>
</div>
<div id="introduction" class="slide section level4">
<h1>Introduction</h1>
<ul>
<li>Material</li>
</ul>
<p>RNA-seq results from amiR396 (miR396 overexpressed) and STTM396 (miR396 being silenced) tomato fruits</p>
<ul>
<li>Objective</li>
</ul>
<p>Visualization of RNA-seq data, especially extract differentially expressed genes I am interested in.</p>
</div>
<div id="overview" class="slide section level4">
<h1>Overview</h1>
<ol style="list-style-type: decimal">
<li>Import CVS file</li>
<li>Evaluation the quality of RNA-seq results (histogram)</li>
<li>Display how many genes are differentially expressed (how many up- or down-regulated).</li>
<li>Display genes with annotation containing targeted words</li>
</ol>
</div>
<div id="load-the-libraries-and-read-csv-file" class="slide section level4">
<h1>Load the libraries and read CSV file</h1>
<pre><code>#load the libraries
import matplotlib.pyplot as plt
import seaborn as sns
import os
import pandas as pd
import numpy as np

#Read CVS file form local
df = pd.read_csv(&#39;/Users/haipingliu/Desktop/BiOF309/Final_project/
DEG_anno.csv&#39;)</code></pre>
</div>
<div id="display-how-many-genes-are-differentially-expressed" class="slide section level4">
<h1>Display how many genes are differentially expressed</h1>
<p><img src="image\gene_N_columns.png" width="1200"></p>
</div>
<div id="display-how-many-genes-have-fold-change-above-2" class="slide section level4">
<h1>Display how many genes have fold change above 2</h1>
<p><img src="image\N_Fold2.png" width="1000"> <img src="image\Head_Fold2.png" width="600"></p>
</div>
<div id="display-up-regulated-genes-in-sttm396-fruits" class="slide section level4">
<h1>Display up-regulated genes in STTM396 fruits</h1>
<p><img src="image\OE_N.png" width="1200"> <img src="image\OE_head.png" width="1200"></p>
</div>
<div id="display-down-regulated-genes-in-sttm396-fruits" class="slide section level4">
<h1>Display down-regulated genes in STTM396 fruits</h1>
<p><img src="image\D_N.png" width="1200"> <img src="image\D_head.png" width="1200"></p>
</div>
<div id="rna-seq-result-follows-standard-normal-distribution-histogram" class="slide section level4">
<h1>RNA-seq result follows standard normal distribution (histogram)</h1>
<p><img src="image\his_distribution.png" width="700"></p>
</div>
<div id="extract-interested-genes-related-with-sweetness-of-fruits" class="slide section level4">
<h1>Extract interested genes related with sweetness of fruits</h1>
<ul>
<li>Sugar <img src="image\Sugar.png" width="1300"></li>
</ul>
</div>
<div id="starch" class="slide section level4">
<h1>Starch</h1>
<p><img src="image\Starch.png" width="1300"></p>
</div>
<div id="sucrose" class="slide section level4">
<h1>Sucrose</h1>
<p><img src="image\Sucrose.png" width="1200"></p>
</div>
<div id="cobine-all-the-genes-related-with-sweetness" class="slide section level4">
<h1>Cobine all the genes related with sweetness</h1>
<p><img src="image\combine_code.png" width="1200"> <img src="image\combined_genes.png" width="800"></p>
</div>
<div id="write-a-csv-file-from-dataframe" class="slide section level4">
<h1>Write a CSV file from DataFrame</h1>
<pre><code>Sugar_gene.to_csv(&#39;Sugar_gene.csv&#39;)</code></pre>
<p><img src="image\sugar_genes_CVS.png" width="1200"></p>
</div>
<div id="get-targeted-columns-fpkm-values-for-two-samples-from-sugar_gene" class="slide section level4">
<h1>Get targeted columns (FPKM values for two samples) from “Sugar_gene”</h1>
<p><img src="image\FPKM.png" width="1200"></p>
</div>
<div id="heatmap-for-expression-of-sweetness-related-genes" class="slide section level4">
<h1>Heatmap for expression of sweetness-related genes</h1>
<pre><code>Heatmap for expression of sweetness-related genes (codes)

df_FPKM_index = df_FPKM.set_index(&#39;Gene_ID&#39;)
sns.clustermap(df_FPKM_index,
              z_score=1,
              metric=&quot;euclidean&quot;,
              method=&quot;ward&quot;,
              cmap=&quot;Blues&quot;,
              annot=True,
              linewidths=1.0)
plt.show()</code></pre>
</div>
<div id="output-for-heatmap" class="slide section level4">
<h1>Output for heatmap</h1>
<p><img src="image\Heatmap.png" width="700"></p>
</div>
<div id="conclusion" class="slide section level4">
<h1>Conclusion</h1>
<ol style="list-style-type: decimal">
<li>RNA-seq result is fine</li>
<li>18 genes related with fruit sweetness are differentially expressed between amiR396 and STTM396 transgenic fruits.</li>
</ol>
</div>
</body>
</html>
